activerecord/lib/active_record/attribute.rb

```diff
@@ -93,7 +93,7 @@ def came_from_user?
     end
 
     def has_been_read?
-      defined?(@value)
+      instance_variable_defined?(:@value)
```

Comment:
`has_been_read?` is being used in the return on `changed_in_place?` and the return of `changed_in_place?` is being used in the return on `changed?`